src/AspNet.Security.OAuth.ArcGIS/ArcGISAuthenticationHandler.cs

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.Extensions.Internal;
+using Newtonsoft.Json.Linq;
+using System;
+
+namespace AspNet.Security.OAuth.ArcGIS {
+    public class ArcGISAuthenticationHandler : OAuthHandler<ArcGISAuthenticationOptions> {
+        public ArcGISAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint + "&token=" + Uri.EscapeDataString(tokens.AccessToken));
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                        
+            // Request the token 
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            response.EnsureSuccessStatusCode();
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+            var error = ArcGISAuthenticationHelper.GetError(payload); // error responses still return 200 status codes
```

Comment:
lol, crazy :smile: 

If the `error` payload contains useful data, consider logging them.